src/mobu/business/base.py

```diff
@@ -17,9 +19,30 @@
 __all__ = ["Business"]
 
 
+class BusinessCommand(Enum):
+    """Commands sent over the internal control queue."""
+
+    STOP = "STOP"
+
+
 class Business:
     """Base class for monkey business (one type of repeated operation).
 
+    The basic flow for a monkey business is as follows:
+
+    - Run ``startup``
+    - In a loop, run ``execute`` followed by ``idle`` until told to stop
+    - When told to stop, run ``shutdown``
+
+    Subclasses should override ``startup``, ``execute``, and ``shutdown`` to
+    add appropriate behavior.  ``idle`` by default waits for ``idle_time``,
+    which generally does not need to be overridden.  Subclasses should also
+    override ``close`` to shut down any object state created in ``__init__``.
+
+    All delays should be done by calling ``pause``, and the caller should
```

Comment:
Relying on the caller to do this on each delay still feels gross, but it's definitely a step forward.

Reply:
Yeah, I was trying to come up with a good alternative but none of them felt very satisfying. I think the next best option would be to raise an exception that's caught by the base business class, but there will still be business-specific cleanup that has to be done when shutting down, so there will need to be a business method that's then called to do that. Although I guess that's `shutdown`, so maybe that's enough. Maybe I'll experiment with that, although it does feel like using exceptions for normal control flow, which is something you're not supposed to do.